File: branches/Medusa/Source/Fantasy.Jobs/IStateBag.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Fantasy.Jobs
{
    public interface IStateBag : IEnumerable<StateBagItem>
    {
        object this[string name]{get; set;}

        int Count { get; }

        string[] Names { get; }

        void Remove(string name);

        bool ContainsState(string name);

        void Clear();
    }

}

File: branches/Stheno/Fantasy.Jobs/src/fantasy/jobs/IStateBag.java
﻿package fantasy.jobs;

public interface IStateBag extends Iterable<StateBagItem>
{
Object getItem(String name);
	void setItem(String name, Object value);

	int getCount();

	String[] getNames();

	void Remove(String name);

	boolean ContainsState(String name);

	void Clear();
}